browser/brave_wallet/brave_wallet_reset.cc

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_wallet/brave_wallet_reset.h"
+#include "brave/browser/brave_wallet/brave_wallet_service_factory.h"
+#include "brave/browser/brave_wallet/eth_tx_controller_factory.h"
+#include "brave/browser/brave_wallet/keyring_controller_factory.h"
+#include "brave/browser/brave_wallet/rpc_controller_factory.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_service.h"
+#include "brave/components/brave_wallet/browser/eth_json_rpc_controller.h"
+#include "brave/components/brave_wallet/browser/eth_tx_controller.h"
+#include "brave/components/brave_wallet/browser/keyring_controller.h"
+
+namespace brave_wallet {
+
+void ResetWallet(content::BrowserContext* context) {
+  auto* eth_tx_controller =
+      brave_wallet::EthTxControllerFactory::GetControllerForContext(context);
+  eth_tx_controller->Reset();
+
+  auto* rpc_controller =
+      brave_wallet::RpcControllerFactory::GetControllerForContext(context);
+  rpc_controller->Reset();
```

Comment:
is it ok that this call will fire `FireNetworkChanged` at same time when service and keyring controller will be dropped ?  maybe make a single entry point and subscribe to KeyringResetFired instead of consecutive calls call?

Reply:
Yes I want `FireNetworkChanged` to be called in particular because there are some members of `eth_tx_state_manager.cc` needs to have that event to reset properly.